packages/firestore/src/platform_browser/browser_connectivity_monitor.ts

```diff
@@ -0,0 +1,63 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { debug } from '../util/log';
+import {
+  ConnectivityMonitor,
+  NetworkStatus
+} from './../remote/connectivity_monitor';
+
+const LOG_TAG = 'ConnectivityMonitor';
+
+/**
+ * Browser implementation of ConnectivityMonitor.
+ */
+export class BrowserConnectivityMonitor implements ConnectivityMonitor {
+  private callbacks: Array<{ (status: NetworkStatus): void }> = [];
+
+  constructor() {
+    this.configureNetworkMonitoring();
+  }
+
+  addCallback(callback: (status: NetworkStatus) => void): void {
+    this.callbacks.push(callback);
+  }
+
+  shutdown(): void {
+    window.removeEventListener('online', this.onNetworkAvailable);
```

Comment:
What's the value of `this` in the callback in this case? Don't you need to do something like: `this.onNetworkAvailable.bind(this)` to make that bind correctly?

Also, I thought I heard that we prefer anonymous functions over binding like this. @mikelehen do you recall? I see plenty of `bind` still in the codebase.

Reply:
It seems like anonymous functions are faster than bind based on this [benchmark](https://jsperf.com/bind-vs-self-closure).

Reply:
I think there isn't going to be a big enough perf difference to warrant weighing that in our decision. We should choose based on readability / consistency with our code.  I have no strong preference at all. I think we do both in our codebase. I think I _slightly_ prefer a closure (`() => this.onNetworkAvailable()`) over bind (`this.onNetworkAvailable.bind(this)`) for readability sake. So that'd be my vote.